### DISCUSSION TITLE
Compatible with ordinary markdown?

### DISCUSSION THREAD

**Discussion by ifree** [observablehq/framework#875] [Q&A]
Great work! I'm wondering if it's possible to make Observable framework compatible with ordinary markdown. 
For example, the framework use triple to denote executable code block, which makes it hard to migrate existing markdown files to Observable.
It would be great if there is a explicit way to denote the code block as executable, and the rest of the markdown file can be rendered as usual. eg. 
For inline code blocks
<pre>js`{code}`, ${js``} </pre> 
Multi-line 
<pre>
```observable/js 
//code
```
</pre>
What do you think?

**Comment by ifree** [comment]
Like [obsidian-dataview](https://github.com/blacksmithgu/obsidian-dataview) it's more beneficial to make it compatible with existing markdown files which unlocks a lot of potential use cases.

**Comment by mbostock** [comment]
We considered various alternatives but this is what we decided on, and I don’t think we are likely to change it. You can disable code execution in JavaScript code blocks by setting `run=false`, such as:

````md
```js run=false
1 + 2
```
````

We are also planning on making code block options configurable both at the page level (via front matter) and the project level (via the project config file) [#156](https://github.com/observablehq/framework/issues/156). This would let you disable JavaScript code execution by default, and then you could opt-in to execution like so:

````md
```js run
1 + 2
```
````

**Reply by ifree** [reply]
Thanks for the quick response! I understand the design decision and appreciate your work.
